src/orchestrator/src/lib.rs

```diff
@@ -255,6 +255,18 @@ pub struct ServicePort {
     pub port_hint: u16,
 }
 
+/// Assignments that the orchestrator has made for a process in a service.
+#[derive(Clone, Debug)]
+pub struct ServiceAssignments<'a> {
+    /// For each specified [`ServicePort`] name, a listen address.
+    pub listen_addrs: &'a BTreeMap<String, String>,
+    /// The listen addresses of each peer in the service.
+    ///
+    /// The order of peers is significant. Each peer is uniquely identified by its position in the
+    /// list.
+    pub peer_addrs: &'a [BTreeMap<String, String>],
```

Comment:
`peer_addrs` also includes `listen_addrs` so you might wonder why we don't replace `listen_addrs` with an `index` field. The reason is that the kubernetes orchestrator needs to pass the same config to each process and therefore can't pass `ServiceAssignments` with varying indexes.

We will be able to change to the `peer_addresses` + `index` representation once we create a separate StatefulSet for each process in a multi-process replica.